### PR TITLE
Fix eslint rules and types of tests

### PR DESCRIPTION
**Pull request by Latropos**
## Summary
Since we have introduced **monorepo** code of example application was provided with more restrictive eslint rules, so I have to either fix runtime tests or disable some of the rules. I do both depending on the case.

* Override eslint rule `no-template-curly-in-string`, because of test-each syntax uses string slooking like string template
* Fix all the type errors, refactor `RuntimeTestsApi.ts`, to rewrite types of test & describe
* Fix imports and ignore most of other rules
 
## Test plan
Tested on IOS

### CODE REVIEW

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/RuntimeTestsApi.ts [hunk 1]
```diff
@@ -209,5 +185,5 @@ export async function recordAnimationUpdates() {
 }
 
 export async function stopRecordingAnimationUpdates() {
-  testRunner.stopRecordingAnimationUpdates();
+  await testRunner.stopRecordingAnimationUpdates();
```
Comment (tjzel) [inline apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/RuntimeTestsApi.ts:188]: Was it caught by eslint? Is it how it was intended to work?

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts [hunk 2]
```diff
@@ -127,11 +128,12 @@ export class TestRunner {
       buildSuite,
       testCases: [],
       nestingLevel: (this._currentTestSuite?.nestingLevel || 0) + 1,
+      // eslint-disable-next-line no-unneeded-ternary
```
Comment (tjzel) [inline apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts:131]: Why do we disable that? I assume eslint wants us to use nullish coalescent operator here?

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts [hunk 3]
```diff
@@ -323,6 +328,7 @@ export class TestRunner {
       console.error = newConsoleFuncJS;
       console.warn = newConsoleFuncJS;
 
+      // eslint-disable-next-line @typescript-eslint/unbound-method
```
Comment (tjzel) [inline apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts:331]: Why is this method unbound?

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/types.ts [hunk 4]
```diff
@@ -25,22 +25,22 @@ export type SharedValueSnapshot = {
 export type ComponentRef = MutableRefObject<(Component & { props: { style: Record<string, unknown> } }) | null>;
 
 export enum DescribeDecorator {
-  ONLY = 'ONLY',
-  SKIP = 'SKIP',
-  NONE = 'NONE',
+  ONLY = 'only',
+  SKIP = 'skip',
```
Comment (tjzel) [inline apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/types.ts:29]: Why did you remove `NONE`? How is it related to monorepo & eslint?

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/types.ts [hunk 5]
```diff
@@ -104,6 +104,7 @@ export type TestConfiguration = {
   render: Dispatch<SetStateAction<ReactNode | null>>;
 };
 
+/* eslint-disable no-var */
```
Comment (tjzel) [inline apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/types.ts:107]: This is ok, but I wonder if with newer TypeScript there's actually a nicer way to do it.

File: apps/common-app/src/examples/RuntimeTests/tests/animations/withTiming/arrays.test.tsx [hunk 6]
```diff
@@ -102,6 +101,7 @@ describe('withTiming animation of ARRAY', () => {
       );
       const components = Object.values(COMPONENT_REF).map(refName => getTestComponent(refName));
 
+      // eslint-disable-next-line @typescript-eslint/no-misused-promises
```
Comment (tjzel) [inline apps/common-app/src/examples/RuntimeTests/tests/animations/withTiming/arrays.test.tsx:104]: Why disable that? 👀

File: apps/common-app/src/examples/RuntimeTests/tests/animations/withTiming/objects.test.tsx [hunk 7]
```diff
@@ -85,6 +79,7 @@ describe('withTiming animation of WIDTH', () => {
       await render(<AnimatedComponent startStyle={startStyle} finalStyle={finalStyle} />);
       const component = getTestComponent(COMPONENT_REF);
       await wait(1000);
+      // eslint-disable-next-line @typescript-eslint/no-misused-promises
```
Comment (tjzel) [inline apps/common-app/src/examples/RuntimeTests/tests/animations/withTiming/objects.test.tsx:82]: Why disable that? 👁️ 👃 👁️

File: apps/common-app/src/examples/RuntimeTests/tests/core/useSharedValue.test.tsx [hunk 8]
```diff
@@ -136,7 +136,7 @@ describe('Tests of *****sharedValue*****', () => {
         await render(<ProgressBar progress={index / Presets.numbers.length} />);
       }
     });
-    test.each([BigInt(2), BigInt(-2), BigInt(123456789), BigInt(123456789123456789123456789)])(
+    test.each([BigInt(2), BigInt(-2), BigInt(123456789), BigInt(1234567891234567)])(
```
Comment (tjzel) [inline apps/common-app/src/examples/RuntimeTests/tests/core/useSharedValue.test.tsx:139]: ??

**Comment by piaskowyk** [comment]
The name of this PR is a bit misleading because it does more than just fixing monorepo compatibility

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts (hunk 2 above)
Comment (Latropos) [reply apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts:131]: I disable it because this was already handled in https://github.com/software-mansion/react-native-reanimated/pull/6047

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/RuntimeTestsApi.ts (hunk 1 above)
Comment (Latropos) [reply apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/RuntimeTestsApi.ts:188]: Yes, there should be `await` here, and preciously eslint didn't catch it.

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/RuntimeTestsApi.ts [hunk 9]
```diff
@@ -158,8 +133,9 @@ export async function wait(delay: number) {
   return testRunner.wait(delay);
 }
 
+// eslint-disable-next-line @typescript-eslint/unbound-method
```
[the remark this answers is not in the document]
Comment (Latropos) [reply apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/RuntimeTestsApi.ts:136]: Because we copy class method, so the state is not preserved. We can 

- disable eslint
- change notify function into arrow function
- change notify into a static function and provide option to ignore static unboud functions https://typescript-eslint.io/rules/unbound-method/#ignorestatic in eslint config

Personally I suggest using ignorestatic, because static functions don't rely on class scope and should be safe to use.

File: apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/types.ts (hunk 4 above)
Comment (Latropos) [reply apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/types.ts:29]: Monorepo required me to use correct types. I wanted to use type `Record<DescribeDecorator, ...>`, so I removed `NONE` which is never used (and now I think I shouldn't have added it to this enum).
Also thats why I changed the values of enum keys - they should match object properties.
Of course I could define more new types without modifying existing ones, but this solution seems the nicest.

File: apps/common-app/src/examples/RuntimeTests/tests/animations/withTiming/arrays.test.tsx (hunk 6 above)
Comment (Latropos) [reply apps/common-app/src/examples/RuntimeTests/tests/animations/withTiming/arrays.test.tsx:104]: Because syntax
```tsx
array.foreach(async () => {})
```
is not allowed. I'll replace it with `for (const X of array)` loop.

File: apps/common-app/src/examples/RuntimeTests/tests/core/useSharedValue.test.tsx (hunk 8 above)
Comment (Latropos) [reply apps/common-app/src/examples/RuntimeTests/tests/core/useSharedValue.test.tsx:139]: image

File: apps/common-app/src/examples/RuntimeTests/tests/animations/withTiming/arrays.test.tsx [hunk 10]
```diff
@@ -102,17 +101,19 @@ describe('withTiming animation of ARRAY', () => {
...
       });
       await wait(speed + 200);
-
-      components.forEach(async (component, index) => {
+      let index = 0;
+      for (const component of components) {
         expect(await component.getAnimatedStyle('width')).toBe(finalWidths[index], ComparisonMode.DISTANCE);
-      });
+        index += 1;
+      }
```
Comment (piaskowyk) [inline apps/common-app/src/examples/RuntimeTests/tests/animations/withTiming/arrays.test.tsx:116]: Why we need this change?

**Review by piaskowyk — APPROVED** [review]
ok for me 👍

File: apps/common-app/src/examples/RuntimeTests/tests/animations/withTiming/arrays.test.tsx (hunk 10 above)
Comment (Latropos) [reply apps/common-app/src/examples/RuntimeTests/tests/animations/withTiming/arrays.test.tsx:116]: The syntax `array.foreach(async () => {})` is not allowed, one of eslint rules ban it.
After reading [this page](https://gist.github.com/joeytwiddle/37d2085425c049629b80956d3c618971) I got convincded that `for` loop is a valid way to write this code.

**Review by tjzel — APPROVED** [review]
